tests/testthat/test-utils-callbacks.R

```diff
@@ -0,0 +1,60 @@
+test_that("CallbackManager catches argument mismatches", {
+  callbacks <- CallbackManager$new()
+
+  expect_snapshot(error = TRUE, {
+    callbacks$add("foo")
+    callbacks$add(function(x, y) x + y)
+    callbacks$add(function(x = 1, y) x + y)
+  })
+
+  expect_silent(callbacks$add(function(x, ...) x))
+  expect_silent(callbacks$add(function(x, y = 1) x + y))
+
+  # Callbacks require one argument, but throw standard R error
+  expect_error(callbacks$invoke())
+  expect_error(callbacks$invoke(1, 2))
+})
+
+test_that("CallbackManager invokes callbacks", {
```

Comment:
I'd suggest breaking this up into smaller tests like "CallbackManager invokes callbacks in LIFO order" and "CallbackManager can remove callbacks"